libbpf-rs/src/program.rs

```diff
@@ -436,6 +489,36 @@ impl Program {
         }
     }
 
+    /// Attach this program to a [userspace
+    /// probe](https://www.kernel.org/doc/html/latest/trace/uprobetracer.html).
+    pub fn attach_uprobe(
+        &mut self,
+        retprobe: bool,
+        pid: i32,
+        binary_path: impl AsRef<Path>,
+        func_offset: usize,
+    ) -> Result<Link> {
+        self.attach_uprobe_impl(
+            pid,
+            binary_path.as_ref(),
+            func_offset,
+            Either::Right(retprobe),
```

Comment:
I'm not sure if Either is a common Rust paradigm, but I think it's clearer from a readability standpoint if we just directly call "libbpf_sys::bpf_program__attach_uprobe()" here instead of having an attach_uprobe_impl() internal function.